src/libasr/pass/intrinsic_functions.h

```diff
@@ -1849,6 +1850,46 @@ namespace Ibits {
 
 } // namespace Ibits
 
+namespace Int {
+
+    static ASR::expr_t *eval_Int(Allocator &al, const Location &loc,
+            ASR::ttype_t* arg_type, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+        ASRUtils::ASRBuilder b(al, loc);
+        if (is_integer(*ASRUtils::expr_type(args[0]))) {
+            return b.i_t(ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_n, arg_type);
```

Comment:
```suggestion
            return b.i2i_t(args[0], arg_type);
```
When we are passing an array as input, we cannot directly down_cast into integer. Rather use i2i_t()